paddle/fluid/framework/hogwild_worker.cc

```diff
@@ -112,6 +117,20 @@ void HogwildWorker::BindingDataFeedMemory() {
 void HogwildWorker::CreateDeviceResource(const ProgramDesc &main_prog) {
   CreateThreadScope(main_prog);
   CreateThreadOperators(main_prog);
+#ifdef PADDLE_WITH_MKLDNN
+  auto EnableMKLDNN = [&](void) {
+    VLOG(3) << "use_mkldnn=True";
+    for (size_t bid = 0; bid < main_prog.Size(); ++bid) {
+      auto *block = const_cast<ProgramDesc &>(main_prog).MutableBlock(bid);
+      for (auto *op : block->AllOps()) {
+        if (op->HasAttr("use_mkldnn")) {
+          op->SetAttr("use_mkldnn", true);
+        }
+      }
+    }
+  };
+  if (FLAGS_use_mkldnn) EnableMKLDNN();
```

Comment:
Why using lambda and not just put it's body into the `if` body?
```suggestion
  if (FLAGS_use_mkldnn) {
    VLOG(3) << "use_mkldnn=True";
    for (size_t bid = 0; bid < main_prog.Size(); ++bid) {
      auto *block = const_cast<ProgramDesc &>(main_prog).MutableBlock(bid);
      for (auto *op : block->AllOps()) {
        if (op->HasAttr("use_mkldnn")) {
          op->SetAttr("use_mkldnn", true);
        }
      }
    }
   }
  };
```

Reply:
ok